mmda/models/base_document_predictor.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from abc import abstractmethod
+from typing import Union, List, Dict, Any
+
+from mmda.types.annotation import Annotation
+from mmda.types.document import Document
+
+
+class BaseDocumentPredictor:
+
+    ###################################################################
+    ##################### Necessary Model Variables ###################
+    ###################################################################
+
+    # TODO[Shannon] Add the check for required backends in the future.
+    # So different models might require different backends:
+    # For example, LayoutLM only needs transformers, but LayoutLMv2
+    # needs transformers and Detectron2. It is the model creators'
+    # responsibility to check the required backends.
+    @property
+    @abstractmethod
+    def REQUIRED_BACKENDS(self):
```

Comment:
this doesnt matter.  each model will have its own service defined w/ its own Dockerfile/requirements/etc.

Reply:
yes -- but they need to import something. And I'll try to use HF's [`LazyModule`](https://github.com/huggingface/transformers/blob/76cadb7943c8492ec481f4f3925e9e8793a32c9d/src/transformers/file_utils.py#L1953) to circumvent that issue

Reply:
why is this necessary?  wouldnt each model py-file have its imports defined at the top?